src/neighborhoods/ellipsoid.jl

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    EllipsoidNeighborhood(semiaxes, angles; convention=:TaitBryanExtr)
+
+An ellipsoid neighborhood with `semiaxes` and `angles`. For 2D ellipses,
+there are two semiaxes and one rotation angle. For 3D ellipsoids, there are
+three semiaxes and three rotation angles. Different rotation conventions can be
+passed via the `convention` keyword argument. The list of conventions is
+available in the [aniso2distance](@ref) documentation.
+"""
+struct EllipsoidNeighborhood <: AbstractBallNeighborhood
+  semiaxes::AbstractVector
+  angles::AbstractVector
+  convention::Symbol
+  metric::Mahalanobis
+
+  function EllipsoidNeighborhood(semiaxes, angles; convention=:TaitBryanExtr)
+    metric = aniso2distance(semiaxes, angles, convention=convention)
+    new(semiaxes, angles, convention, metric)
+  end
+end
+
+radius(ellips::EllipsoidNeighborhood) = one(eltype(ellips.semiaxes))
```

Comment:
The radius function doesn't make much sense for the Ellipsoid. Can we remove this here and modify the search method slightly to handle this instead?